main.nf

```diff
@@ -2,26 +2,43 @@
 nextflow.enable.dsl=2
 
 process foo {
-    tag "${name}"
-    echo true
     cpus 1
     memory 500.MB
     container "opensuse/leap:latest"
-    cache 'lenient'
     input:
-        val(name)
-        path(template_file)
+        path input_files
     output:
-        path("${name}.txt")
+        path "files/*.txt", emit: files
     """
-    sha1sum ${template_file}
-    cat ${template_file} > ${name}.txt
-    echo ${name} >> ${name}.txt
+    mkdir files/
+    cp ${input_files} files/
+    """
+}
+
+process bar {
```

Comment:
This is not at all how I would do this. Please change all of this to be more like what I would do.